### PR TITLE
Mccalluc/try prov

### PR DESCRIPTION
**Pull request by mccalluc**
Trying to use HCA Project was not going well. There is a W3C *Prov* ontology that I think does what we need to do, and there is a python package, also called `prov`, which can catch some typos (though I'm not sure it's of much use beyond that). For the five objects we have, I think their relationships could be expressed this way.

Ideally, the IDs in the documents would match the URIs in the RDF, except the HCA schemas require IDs to be formatted like long UUIDs. Down the road, may support that, but for now it would just make our examples less clear. (See #17.)

This includes work in PR #7 that I wanted to build on to avoid merge conflicts.

### CODE REVIEW

**Comment by mruffalo** [comment]
I have some minor suggestions in terms of the code (e.g. using newer packages/idioms), but otherwise this looks reasonable. I'll add some comments to the code changes, but none of those will block merging this.

**Comment by shirey** [comment]
This definitely looks like the right way to go.

File: test.py [hunk 1]
```diff
@@ -19,40 +22,54 @@ def download_to(url, target):
     os.rename(download_path, target)
 
 
+def validate_json(dir_path, name, fails, hubmap_schema):
```
Comment (mruffalo) [inline test.py]: I'm not usually a fan of "returning" data from a function via giving a mutable data structure as input and letting the function modify it, but that can be appropriate at times. I'm not really convinced that this is one of the times in which this is necessary.

File: test.py [hunk 2]
```diff
@@ -19,40 +22,54 @@ def download_to(url, target):
...
+    print(f'\tLoad JSON: {name}')
+    path = os.path.join(dir_path, name)
```
Comment (mruffalo) [inline test.py]: The `pathlib` module provides a much nicer interface for manipulating filesystem paths, and is worth using when possible.

File: test.py (hunk 1 above)
Comment (mccalluc) [reply test.py]: Since this is merged, I filed new issue. #23

File: test.py (hunk 2 above)
Comment (mccalluc) [reply test.py]: Since this is merged, I filed new issue. #22

**Pull request merged**